config.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"os"
+
+	log "github.com/Sirupsen/logrus"
+	flag "github.com/spf13/pflag"
+	config "github.com/spf13/viper"
+)
+
+var (
+	confFile      string
+	logLevel      string
+	restAddress   string
+	rpcAddress    string
+	localStateDir string
+)
+
+var (
+	defaultConfPaths = []string{
+		"/etc/glusterd",
+		".",
+	}
+	defaultConfName = "glusterdconf"
+)
+
+// parseFlags sets up the flags and parses them, this needs to be called before any other operation
+func parseFlags() {
+	flag.StringVar(&confFile, "config", "", "Configuration file for GlusterD. By default looks for glusterdconf.(yaml|toml|json) in /etc/glusterd and current working directory.")
+	flag.StringVar(&localStateDir, "localstatedir", "", "Directory to store local state information. Defaults to current working directory.")
```

Comment:
This was a carry forward from what was there before. Also, as I've added in the comment for `setDefaults` it can be used to setup config options which are not available as flags. I don't expect all config options to be available as flags, even though right now, they are all flags.